src/rpdk/core/contract/suite/hook/hook_handler_commons.py

```diff
@@ -0,0 +1,96 @@
+import logging
+
+from rpdk.core.contract.hook_client import HookClient
+from rpdk.core.contract.interface import HandlerErrorCode, HookStatus
+from rpdk.core.contract.suite.contract_asserts_commons import failed_event
+
+LOG = logging.getLogger(__name__)
+
+UNSUPPORTED_TARGET = "AWS::FakeService::Resource"
+
+
+def _prepare_target_model(invocation_point):
+    target_model = {"resourceProperties": {}}
+    if HookClient.is_update_invocation_point(invocation_point):
+        target_model["previousResourceProperties"] = {}
+
+    return target_model
+
+
+def test_hook_success(hook_client, invocation_point, target, target_model):
+    if HookClient.is_update_invocation_point(invocation_point):
+        raise ValueError(
+            "Invocation point {} not supported for this testing operation".format(
+                invocation_point
+            )
+        )
+
+    _status, response, _error_code = hook_client.call_and_assert(
+        invocation_point, HookStatus.SUCCESS, target, target_model
+    )
+
+    return response
+
+
+def test_update_hook_success(hook_client, invocation_point, target, target_model):
+    if not HookClient.is_update_invocation_point(invocation_point):
+        raise ValueError(
+            "Invocation point {} not supported for testing UPDATE hook operation".format(
+                invocation_point
+            )
+        )
+
+    _status, response, _error_code = hook_client.call_and_assert(
+        invocation_point, HookStatus.SUCCESS, target, target_model
+    )
+
+    return response
+
+
+def test_hook_failed(hook_client, invocation_point, target, target_model=None):
+    _status, response, error_code = hook_client.call_and_assert(
+        invocation_point, HookStatus.FAILED, target, target_model
+    )
+    assert response["message"]
+    return response, error_code
+
+
+@failed_event(
+    error_code=HandlerErrorCode.InvalidRequest,
+    msg="A hook handler MUST return FAILED with a InvalidRequest error code if the target is not supported",
```

Comment:
Similar to https://github.com/aws-cloudformation/cloudformation-cli/issues/527#issuecomment-680696823, since `targetNames` are already explicitly modeled in the schema, seems redundant to also require handling that in hook handlers. Think the platform itself should handle only invoking relevant hook handlers based on their `targetNames` (think this contract can just be removed if that's already the case)

Reply:
This is no longer being used, have removed it in latest revision